### PR TITLE
Add restore-point-list command for gp

### PR DESCRIPTION
**Pull request by rogaliiik**
Greenplum

Add restore-point-list command for Greenplum

### CODE REVIEW

**Comment by rogaliiik** [comment]
The functionality of this command is very similar to "backup-list" command. Should i write unittests for these functions?

File: utility/utility.go [hunk 1]
```diff
@@ -225,6 +228,16 @@ func stripBackupSuffix(pathValue string) string {
...
 }
 
+func StripRightmostRestorePointName(path string) string {
+	path = strings.Trim(path, "/")
+	all := strings.SplitAfter(path, "/")
+	return stripRestorePointSuffix(all[len(all)-1])
+}
+
+func stripRestorePointSuffix(pathValue string) string {
+	return strings.Split(pathValue, "_restore_point")[0]
```
Comment (usernamedt) [inline utility/utility.go:238]: use the `RestorePointSuffix` constant

File: internal/restore_point_list_handler.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package internal
```
Comment (usernamedt) [inline internal/restore_point_list_handler.go:1]: Restore point logic is greenplum-specific and should belong to the `internal/databases/greenplum`.

File: cmd/gp/restore_point_list.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/wal-g/wal-g/utility"
+)
+
+const (
+	restorePointListShortDescription = "Prints available restore points"
+)
+
+var (
+	// restorePointListCmd represents the restorePointList command
+	restorePointListCmd = &cobra.Command{
```
Comment (usernamedt) [inline cmd/gp/restore_point_list.go:16]: Add the corresponging notes to the documentation.

File: utility/utility.go [hunk 4]
```diff
@@ -58,6 +58,9 @@ const (
 	StreamMetadataFileName = "stream_metadata.json"
```
Comment (usernamedt) [inline utility/utility.go:58]: Same, this logic is greenplum-specific.

**Review by usernamedt — CHANGES_REQUESTED** [review]

**Comment by usernamedt** [comment]
> The functionality of this command is very similar to "backup-list" command. Should i write unittests for these functions?

A couple of unit test that checks that command works as expected will be enough

**Review by usernamedt — APPROVED** [review]

**Pull request merged**